### PR TITLE
Input support for OpenXR

### PR DESCRIPTION
**Pull request by Manishearth**
Requires https://github.com/servo/webxr/pull/65

This treats the right hand as an input source. While I've tested the input code, I haven't yet tested this hooked up to the Servo DOM (https://github.com/servo/servo/pull/24362), so I'd like to hold off on merging this until I get a chance

Still, feel free to review the last few commits that have "openxr" in their bodies

### CODE REVIEW

**Comment by Manishearth** [comment]
This works. r? @jdm @asajeffrey

File: webxr/openxr/mod.rs [hunk 1]
```diff
@@ -436,10 +440,23 @@ impl Device for OpenXrDevice {
             .locate(&self.space, self.frame_state.predicted_display_time)
             .unwrap();
 
+        let pose_valid = location.location_flags.intersects(
+            SpaceLocationFlags::ORIENTATION_VALID | SpaceLocationFlags::ORIENTATION_VALID,
```
Comment (jdm) [inline webxr/openxr/mod.rs]: This looks weird - are there supposed to be two different flags here?

Comment (Manishearth) [reply webxr/openxr/mod.rs]: Yes

Comment (Manishearth) [reply webxr/openxr/mod.rs]: Fixed.

**Pull request merged**